pymc3/distributions/distribution.py

```diff
@@ -420,6 +419,20 @@ def draw_values(params, point=None, size=None):
                     evaluated[param_idx] = drawn[(param, size)]
                 else:
                     try:  # might evaluate in a bad order,
+                        # Sometimes _draw_value recurrently calls draw_values.
+                        # This may set values for certain nodes in the drawn
+                        # dictionary, but they don't get added to the givens
+                        # dictionary. Here, we try to fix that.
+                        if param in named_nodes_children:
```

Comment:
From my cursory knowledge this looks good! Out of curiosity though why would the bug only present itself in one version of python?

Reply:
I was super puzzled by this too! I finally found that the problem was caused because python 3.6 [preserves insertion](https://stackoverflow.com/a/39980744/7253166) order for sets and dictionaries, whereas earlier versions did not.

Reply:
Ah yes, optional but related. Raymond Hettinger gave an excellent talk on how this came about
https://www.youtube.com/watch?v=npw4s1QTmPg and he was nice enough to answer my question about it on Stackoverflow!

https://stackoverflow.com/questions/44172039/in-raymond-hettingers-pycon-2017-talk-what-is-the-database-representation